File: data/gencode/ItemCfg.go
package gencode

import (
	"encoding/json"
	"gs/define"
	"gs/lib/mylog"
	"io/ioutil"
	"os"
	"sync"
)

type ItemCfg struct {
	ItemSlc []*ItemInfo `json:"Item"`
	ItemMap map[int32]*ItemInfo
}

type ItemInfo struct {
	ID      int32  // 物品id
	Name    string // 物品名字
	Type    int32  // 物品类型
	Quality int32  // 品质
}

var itemCfg *ItemCfg
var itemOnce sync.Once

func GetItemCfg() *ItemCfg {
	itemOnce.Do(func() {
		itemCfg = new(ItemCfg)
		itemCfg.init()
	})
	return itemCfg
}

func (this *ItemCfg) init() {
	this.ItemMap = make(map[int32]*ItemInfo)
	rootPath := os.Getenv(define.EnvName)
	filePtr, err := os.Open(rootPath + "/data/json/Item.json")
	if err != nil {
		mylog.Error("load ItemCfg failed", err)
		return
	}
	defer filePtr.Close()
	data, err := ioutil.ReadAll(filePtr)
	if err != nil {
		mylog.Error("load ItemCfg failed", err)
		return
	}
	err = json.Unmarshal(data, this)
	if err != nil {
		mylog.Error("load ItemCfg failed", err)
		return
	}
	for _, v := range this.ItemSlc {
		this.ItemMap[v.ID] = v
	}
}

func (this *ItemCfg) GetItemById(id int32) (*ItemInfo, bool) {
	if ret, ok := this.ItemMap[id]; ok {
		return ret, ok
	}
	return nil, false
}

File: data/gencode/ItemCfg.ts
import Res from "../../common/util/Res"

export class ItemInfo {
    public ID: number// 物品id
    public Name: string// 物品名字
    public Type: number// 物品类型
    public Quality: number// 品质
}

export class ItemConfig {
    private static instance: ItemConfig
    public static Get(): ItemConfig {
        if (ItemConfig.instance == null) {
            ItemConfig.instance = new ItemConfig()
            ItemConfig.instance.init()
        }
        return ItemConfig.instance
    }

    public ItemSlc: Array<ItemInfo>
    public ItemMap: Map<number, ItemInfo>

    private init(): void {
        this.ItemSlc = new Array<ItemInfo>()
        this.ItemMap = new Map<number, ItemInfo>()

        let jsonData = Res.get<cc.JsonAsset>("json/Item", cc.JsonAsset)
        this.ItemSlc = jsonData.json['Item']
        this.ItemSlc.forEach(item => {
            this.ItemMap.set(item.ID, item)
        })

    }
}
